### PR TITLE
CV2-4998: extract link as tipline message

### PR DESCRIPTION
**Pull request by melsawy**
## Description

For tipline messages extract a message that contain a link as an item of type Link.

References: CV2-4998

## How has this been tested?

Implemented automated tests.

## Things to pay attention to during code review

Please describe parts of the change that require extra attention during code review, for example:

- File FFFF, line LL: This refactoring does this and this. Is it consistent with how it’s implemented elsewhere?
- Etc.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have added unit and feature tests, if the PR implements a new feature or otherwise would benefit from additional testing
- [ ] I have added regression tests, if the PR fixes a bug
- [ ] I have added logging, exception reporting, and custom tracing with any additional information required for debugging
- [ ] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] If I added a third party module, I included a rationale for doing so and followed our current [guidelines](https://meedan.atlassian.net/wiki/spaces/ENG/overview#Choose-the-%E2%80%9Cright%E2%80%9D-3rd-party-module)

### CODE REVIEW

File: app/models/concerns/smooch_messages.rb [hunk 1]
```diff
@@ -291,7 +291,8 @@ def bundle_list_of_messages_to_items(list, last)
         if message['type'] == 'text'
           # Get an item for long text (message that match number of words condition)
           if message['payload'].nil?
-            messages << message if ::Bot::Alegre.get_number_of_words(message['text'].to_s) > CheckConfig.get('min_number_of_words_for_tipline_submit_shortcut', 10, :integer)
+            contain_link = self.extract_url(message['text'])
+            messages << message if !contain_link.nil? || ::Bot::Alegre.get_number_of_words(message['text'].to_s) > CheckConfig.get('min_number_of_words_for_tipline_submit_shortcut', 10, :integer)
```
Comment (caiosba) [inline app/models/concerns/smooch_messages.rb:295]: For consistency, please just rename `contain_link` to `contains_link`.

**Review by caiosba — APPROVED** [review]
Thanks Sawy, just one minor change before merging. And nice test added!

File: app/models/concerns/smooch_messages.rb (hunk 1 above)
Comment (melsawy) [reply app/models/concerns/smooch_messages.rb]: Done

**Comment by melsawy** [comment]
@caiosba I used `Twitter::TwitterText::Extractor.extract_urls` instead of `extract_url` as we only need to know if contains url or not and `extract_url` handle other cases which not required here.

**Comment by caiosba** [comment]
> @caiosba I used `Twitter::TwitterText::Extractor.extract_urls` instead of `extract_url` as we only need to know if contains url or not and `extract_url` handle other cases which not required here.

OK, sounds good!

**Pull request merged**